src/materialized/src/bin/materialized/main.rs

```diff
@@ -388,6 +388,10 @@ pub struct Args {
     #[clap(long, env = "MZ_CLUSTER_REPLICA_SIZES")]
     cluster_replica_sizes: Option<String>,
 
+    /// Availability zones compute resources may be deployed in.
+    #[clap(long, env = "MZ_ALLOW_AVAILABILITY_ZONE", use_value_delimiter = true)]
+    allow_availability_zone: Vec<String>,
```

Comment:
Bike-shedding:
1. Should we drop the `allow-` prefix, like we did for `cluster-replica-sizes`?
2. We use `use_value_delimiter = true` to also support passing in multiple values via env variable too. AFAICS, there is no way to tell clap to only use delimiters for values read from the env, so it is now possible to do both:
    - `--allow-availability-zone us-east-1a --allow-availability-zone us-east-1b` and
    - `--allow-availability-zone us-east-1a,us-east-1b`
    
    Alternatively, we could just drop the `MZ_ALLOW_AVAILABILITY_ZONE` support, if it doesn't seem helpful.

Reply:
On 1., I'm in favor or skipping the `allow-` prefix. It adds less clutter.

Reply:
Should we make sure that the list of availability zones is unique?

Reply:
Uniqueness doesn't really matter for checking replica AZs, but I guess enforcing it could help discover misconfiguration by clients of the CLI. I'll add a check.